Validation/RecoB/plugins/BDHadronTrackMonitoringAnalyzer.cc

```diff
@@ -0,0 +1,458 @@
+#include "Validation/RecoB/plugins/BDHadronTrackMonitoringAnalyzer.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h" 
+
+#include "DQMOffline/RecoB/interface/Tools.h"
+#include "DataFormats/GeometryVector/interface/GlobalVector.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+#include "TrackingTools/Records/interface/TransientTrackRecord.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateOnSurface.h"
+#include "TrackingTools/IPTools/interface/IPTools.h"
+#include "RecoVertex/VertexPrimitives/interface/ConvertToFromReco.h"
+
+#include <iostream>
+#include <fstream>
+
+using namespace reco;
+using namespace edm;
+using namespace std;
+
+
+std::map<unsigned int, std::string> TrkHistCat{
```

Comment:
@smoortga -please move this into a header where it can be properly shared (eg, this link has some examples that should work http://stackoverflow.com/questions/23504339/creating-and-populating-a-map-in-a-header-file-in-c)